packages/solana/src/deposit.ts

```diff
@@ -221,6 +221,19 @@ const tryToAlignDepositsAndTransfers = (
   return results;
 };
 
+export class TransactionMatchingError extends Error {
+  constructor(
+    message: string,
+    public cause: Error,
+    public deposits: DepositInfo[],
+    public transfers: Transfer[],
+  ) {
+    super(message);
+    this.name = TransactionMatchingError.name;
+    Error.captureStackTrace(this, TransactionMatchingError);
```

Comment:
i have no idea how this works, but shouldnt the `super(message)` call already do this?

Reply:
I think the main difference is that it erases the constructor call from the stack trace. Honestly I've been doing it for so long idk why anymore